lib/App/MintTag.pm

```diff
@@ -493,6 +495,59 @@ sub _maybe_delete_source_branches ($self, $step) {
   }
 }
 
+sub _maybe_cleanup_tags ($self, $step) {
+  return unless $step->push_tag_to && $step->cleanup_tag_days;
+
+  my $remote = $step->push_tag_to;
+  my $prefix = $step->tag_prefix;
+
+  try {
+    my $tag_spec = "refs/tags/${prefix}*";
+    my $tag_list = run_git('for-each-ref', "--format=%(refname:short)\t%(creatordate:unix)", $tag_spec);
+
+    my @tags = map {;
+      my ($tag, $ts) = split /\t/;
+      { ts => $ts, 'tag' => $tag}
+    } split /\n/, $tag_list;
+
+    my $time = time;
+
+    my $max_ts = $time - ($step->cleanup_tag_days * 86400);
+
+    my @can_remove = map {; $_->{tag} } grep {; $_->{ts} < $max_ts } @tags;
+
+    my $tag_list_count = @tags;
+    my $tag_remove_count = @can_remove;
+
+    # if all tags except the one we've just tagged are old enough to remove,
+    # make sure to keep the latest tag by removing it from the deletion list
+    if ($tag_remove_count > 0 and $tag_remove_count >= $tag_list_count - 1) {
```

Comment:
Thanks. I think this works. One thing to consider: for-each-ref is returning a sort based on tag name. Our tag names typically sort correct because the way we do them. But what if our latest tag is a point release on top of an older date's tag? We'll keep the wrong latest tag. It probably doesn't matter. But one way to prevent that is to sort based on ts:
```
@tags = sort { $a->{ts} <=> $b->{ts} } @tags;
```